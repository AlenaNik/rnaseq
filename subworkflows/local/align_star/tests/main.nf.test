nextflow_workflow {

    name "Test Subworkflow ALIGN_STAR"
    script "../main.nf"
    workflow "ALIGN_STAR"
    config "./nextflow.config"

    tag "ALIGN_STAR"

    tag "star/align"
    tag "star/genomegenerate"

    tag "STAR_ALIGN_IGENOMES"
    tag "STAR_GENOMEGENERATE_IGENOMES"

    tag "BAM_SORT_STATS_SAMTOOLS"

    test("star - no igenomes") {

        setup {
            run("STAR_GENOMEGENERATE") {
                script "../../../../modules/nf-core/star/genomegenerate/main.nf"
                process {
                    """
                    input[0] = Channel.of([
                        [ id:'test_fasta' ],
                        [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
                    ])
                    input[1] = Channel.of([
                        [ id:'test_gtf' ],
                        [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                    ])
                    """
                }
            }
        }

        when {
            workflow {
                """
                star_ignore_sjdbgtf = false
                seq_platform        = 'illumina'
                seq_center          = false
                is_aws_igenome      = false

                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = STAR_GENOMEGENERATE.out.index
                input[2] = Channel.of([
                    [ id:'test_gtf' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                input[3] = star_ignore_sjdbgtf
                input[4] = seq_platform
                input[5] = seq_center
                input[6] = is_aws_igenome
                input[7] = Channel.of([
                    [ id:'test_fasta' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
    test("star - with igenomes") {

        setup {
            run("STAR_GENOMEGENERATE_IGENOMES") {
                script "../../../../modules/local/star_genomegenerate_igenomes/main.nf"
                process {
                    """
                    input[0] = Channel.of([file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)])
                    input[1] = Channel.of([file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true)])
                    """
                }
            }
        }

        when {
            workflow {
                """
                star_ignore_sjdbgtf = false
                seq_platform        = 'illumina'
                seq_center          = false
                is_aws_igenome      = true

                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = STAR_GENOMEGENERATE_IGENOMES.out.index.map{ [ [id:'star'], it ] }
                input[2] = Channel.of([
                    [ id:'test_gtf' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                input[3] = star_ignore_sjdbgtf
                input[4] = seq_platform
                input[5] = seq_center
                input[6] = is_aws_igenome
                input[7] = Channel.of([
                    [ id:'test_fasta' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

}
