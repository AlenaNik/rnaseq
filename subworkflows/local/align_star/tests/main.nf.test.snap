{
    "star - with igenomes": {
        "content": [
            "test.Log.final.out",
            "test.Log.out",
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam.bai:md5,8bfbd4094a92ac0de8492d98a471d1b0"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam:md5,fc154c31eb4d034ebdb6b1d17c6b45eb"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Aligned.sortedByCoord.out.bam:md5,b6cec9bba6b04b9b92eddbef128bdfbb"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Log.progress.out:md5,b2bd061d6cbaaf3d6d3b1fed547f69b8"
                ]
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.flagstat:md5,db0e25cd0b37d3030e807846c022199e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.idxstats:md5,107ca94dd426cc44db316f0d402307c5"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Aligned.sortedByCoord.out.bam:md5,b6cec9bba6b04b9b92eddbef128bdfbb"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.stats:md5,202e2f234f93b347b6a72d0163773054"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.SJ.out.tab:md5,844af19ab0fc8cd9a3f75228445aca0d"
                ]
            ],
            [
                "versions.yml:md5,06693106908e6f8f38a2c30accf7067d",
                "versions.yml:md5,369619588c8c294b74dca9058a151b11",
                "versions.yml:md5,78069d4515b04251f758ac52a49d9971",
                "versions.yml:md5,b6a22ef369a375445a8f9313721b8092",
                "versions.yml:md5,e3b043ba840d5bddde92f0bf1c91a5fc",
                "versions.yml:md5,feaf798d368c662a7a1cea87a24a7434"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-03T10:42:50.176094"
    },
    "star - no igenomes - stub": {
        "content": [
            "test.Log.final.out",
            "test.Log.out",
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test.Aligned.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.toTranscriptome.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam.csi:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Log.progress.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "/home/max/tests/6e361e57e6a3bdc7a9658b65cbbacd6/work/15/f5d31a04cce8dd4fdb0d695c29d4b7/test.unmapped_1.fastq.gz",
                        "/home/max/tests/6e361e57e6a3bdc7a9658b65cbbacd6/work/15/f5d31a04cce8dd4fdb0d695c29d4b7/test.unmapped_2.fastq.gz"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.flagstat:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.idxstats:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test.Aligned.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "testXd.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.stats:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test.ReadsPerGene.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test.SJ.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test.tab:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            ],
            [
                "versions.yml:md5,06693106908e6f8f38a2c30accf7067d",
                "versions.yml:md5,35c1cc21d26d9d8f00e6ec87e97fb634",
                "versions.yml:md5,369619588c8c294b74dca9058a151b11",
                "versions.yml:md5,78069d4515b04251f758ac52a49d9971",
                "versions.yml:md5,b6a22ef369a375445a8f9313721b8092",
                "versions.yml:md5,feaf798d368c662a7a1cea87a24a7434"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-03T10:43:10.186065"
    },
    "star - no igenomes": {
        "content": [
            "test.Log.final.out",
            "test.Log.out",
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam.bai:md5,8bfbd4094a92ac0de8492d98a471d1b0"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam:md5,e6573d093da6ec95f99c1669eb86bbd6"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Aligned.sortedByCoord.out.bam:md5,b9ee1c607e07323bc1652ef3babb543f"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Log.progress.out:md5,b2bd061d6cbaaf3d6d3b1fed547f69b8"
                ]
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.flagstat:md5,db0e25cd0b37d3030e807846c022199e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.idxstats:md5,107ca94dd426cc44db316f0d402307c5"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Aligned.sortedByCoord.out.bam:md5,b9ee1c607e07323bc1652ef3babb543f"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.stats:md5,202e2f234f93b347b6a72d0163773054"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.SJ.out.tab:md5,844af19ab0fc8cd9a3f75228445aca0d"
                ]
            ],
            [
                "versions.yml:md5,06693106908e6f8f38a2c30accf7067d",
                "versions.yml:md5,35c1cc21d26d9d8f00e6ec87e97fb634",
                "versions.yml:md5,369619588c8c294b74dca9058a151b11",
                "versions.yml:md5,78069d4515b04251f758ac52a49d9971",
                "versions.yml:md5,b6a22ef369a375445a8f9313721b8092",
                "versions.yml:md5,feaf798d368c662a7a1cea87a24a7434"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-03T10:41:54.293096"
    },
    "star - with igenomes - stub": {
        "content": [
            "test.Log.final.out",
            "test.Log.out",
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test.Aligned.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.toTranscriptome.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam.csi:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Log.progress.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "/home/max/tests/6efdcb422a49d8bf15b844d59e0dc8a1/work/fa/b5d0173593fa74664b340bb2e08e07/test.unmapped_1.fastq.gz",
                        "/home/max/tests/6efdcb422a49d8bf15b844d59e0dc8a1/work/fa/b5d0173593fa74664b340bb2e08e07/test.unmapped_2.fastq.gz"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.flagstat:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.idxstats:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test.Aligned.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "testXd.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.stats:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test.ReadsPerGene.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test.SJ.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test.tab:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            ],
            [
                "versions.yml:md5,06693106908e6f8f38a2c30accf7067d",
                "versions.yml:md5,369619588c8c294b74dca9058a151b11",
                "versions.yml:md5,78069d4515b04251f758ac52a49d9971",
                "versions.yml:md5,b6a22ef369a375445a8f9313721b8092",
                "versions.yml:md5,e3b043ba840d5bddde92f0bf1c91a5fc",
                "versions.yml:md5,feaf798d368c662a7a1cea87a24a7434"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-03T10:43:34.421071"
    }
}